engine/gui/widgets.py

```diff
@@ -1,4 +1,6 @@
+from collections import Counter
```

Comment:
Style: import modules, not classes/functions: https://google.github.io/styleguide/pyguide.html#22-imports

So here you'd do:
```python
import collections

# ...

c = collections.Counter()
```
There are a few exceptions to this rule, such as things that are used for static types.

Repeat it for the `pathlib` stuff too.

Reply:
Yeah this code is currently wrong; `selected_button` should _always_ exist. We'll want to get rid of the `Optional` aspect of it.

Instead we'll want to error when one of these conditions is true:
* `initial_selected_button` is not specified.
* There is no button with the `name` field matching `initial_selected_button`.

Most of the code is already there, the problem is that we silently continue on if either of those conditions are true. We should instead raise a `ValidationError`.